cluster-autoscaler/debuggingsnapshot/debugging_snapshotter_test.go

```diff
@@ -0,0 +1,160 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package debuggingsnapshot
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/pkg/scheduler/framework"
+)
+
+func TestBasicSnapshotRequest(t *testing.T) {
+	var wg sync.WaitGroup
+	wg.Add(1)
+	snapshotter := NewDebuggingSnapshotter(true)
+
+	pod := []*framework.PodInfo{
+		{
+			Pod: &v1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "Pod1",
+				},
+				Spec: v1.PodSpec{
+					NodeName: "testNode",
+				},
+			},
+		},
+	}
+	node := &v1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "testNode",
+		},
+	}
+
+	nodeInfo := &framework.NodeInfo{
+		Pods:             pod,
+		Requested:        &framework.Resource{},
+		NonZeroRequested: &framework.Resource{},
+		Allocatable:      &framework.Resource{},
+		Generation:       0,
+	}
+
+	var nodeGroups []*framework.NodeInfo
+	nodeGroups = append(nodeGroups, nodeInfo)
+	nodeGroups[0].SetNode(node)
+
+	req := httptest.NewRequest(http.MethodGet, "/", nil)
+	w := httptest.NewRecorder()
+
+	go func() {
+		snapshotter.ResponseHandler(w, req)
+		wg.Done()
+	}()
+
+	for !snapshotter.IsDataCollectionAllowed() {
```

Comment:
I know this is only a test, but let's avoid busy waiting and add some sleep() here (and in other tests below) instead of burning cpus.

Reply:
I purposefully avoided using Sleep in these tests. (Did consider Sleep as an alternative). Since the whole test execution is in microseconds I don't really see any substantial gains here.

Also, this loop should ideally only execute once, since the conditional statement is waiting for a lock, that will be freed by ResponseHandler and it should directly be able to execute the command, inside the loop, when the conditional statement fails.

For the more indefinite / unknown time-period wait, I am using a Waitgroup anyways.